Выполнила проектную работу "Карты подскажут".

olga-rks opened the pull request:
Отправляю проектную работу на ревью.

olga-rks committed:
Выполнила проектную работу "Карты подскажут".

Отправляю проектную работу на ревью.

Avaakum commented on index.html:
```diff
@@ -1,22 +1,169 @@
+            href="https://apps.apple.com/us/app/oblique-strategies-se/id1104927011"
+            target="_blank"
+            rel="noopener noreferrer"
+            >iOS</a>
```
Текст в ссылке должен быть "для iOS"

Avaakum commented on index.html:
```diff
@@ -1,22 +1,169 @@
+    <link rel="icon" href="./images/favicon.ico" sizes="any">
+    <link rel="icon" href="./images/favicon.svg" type="image/svg+xml">
+    <link rel="apple-touch-icon" href="./images/favicon.png">
+    <link rel="stylesheet" href="./styles/style.css">
```
---

**Отлично:**

Корректно заполнен `head`

Avaakum commented on index.html:
```diff
@@ -1,22 +1,169 @@
+      <h1 class="heading content-section header__title">Карты подскажут</h1>
+    </header>
+    <nav class="header__menu content-section">
+      <div class="header__logo">
```
---

**Можно лучше:**

Логотип должен быть ссылкой **без атрибута href,**(что бы по ссылке не было перехода). А так же, что бы при навигации с клавиатуры этот элемент не получал видимый фокус. **Пустой атрибут href тоже нужно удалить.**

Avaakum commented on index.html:
```diff
@@ -1,22 +1,169 @@
+      <figure class="content__figure">
+        <picture>
+          <source
+            srcset="./images/cards-1x.avif"
```
Так как у нас по 2 изображения каждого типа(формата), используйте вместо 2х source один с записью источников изображений с одним форматом, но разной плотностью пикселей и с указанием этой плотности пикселей. 

Пример ` <source srcset="640x480.png 1x, 1280x960.png 2x"> ` 

Дополнительно https://doka.guide/html/source/#srcset

Avaakum commented on index.html:
```diff
@@ -1,22 +1,169 @@
+          <source
+            srcset="./images/cards-2x.webp"
+            type="image/webp">
+          <img class="content__picture" src="./images/cards-1x.png" loading="lazy"
```
Так же с srcset нужно реализовать для img

Avaakum commented on index.html:
```diff
@@ -1,22 +1,169 @@
+        <p class="content__paragraph">
+          Вот что сам Брайан Ино рассказывает о создании Oblique Strategies:
+        </p>
+        <blockquote class="content__blockquote">
```
---

**Отлично:**

Отличная семантика

Avaakum reviewed (changes requested):
Развернуть общий комментарий ⬇︎

Доброго времени суток. У вас получилась хорошая работа!

_Что понравилось:_

- работают темы
- отличная структура файлов проекта
- Хорошая семантика
- Хорошие стили 
- Отзывчивая верстка


_Что необходимо исправить:_

- Все комментарии, кроме комментариев «Можно лучше» и «Отлично»

- нет ховера из макета у сслок, должно быть подчёркивание с прозрачностью

- https://disk.yandex.ru/i/B4cfnQ9bedRdkQ  Маркеры списка должны находится по центру  напротив элемента списка, а не по верхней строке. Для этого нужно опустить вниз буллит на  50% элемента списка , например с помощью  inset-block-start и затем поднять его на половину собственной высоты(7.5px) с помощью margin-block-start. 

- При ширине экрана 1440px _не совпадает с макетом_:
   - https://disk.yandex.ru/i/BcmbUCN5heBmDw выаравнивание эл-а

- При ширине экрана 360px _не совпадает с макетом_:
   -    https://disk.yandex.ru/i/K3UsifSt_xZP1w выаравнивание эл-а




_Что можно улучшить:_

- Все комментарии с пометкой "Можно лучше"
- Для всех функций ` var() ` в CSS нужно задать резервные значения,  на случай если доступ к кастомному свойству по имени будет невозможен. 
- Можно использовать атрибуты `target="_blank"` для ссылок на сторонние ресурсы для того чтобы пользователю не приходилось заново открывать сайт проекта
- больше использовать currentColor, где это возможно
- больше использовать vi вместо vw


Желаю успеха!

olga-rks committed:
Исправление ошибок после ревью.

Отправляю проектную работу "Карты подскажут" на ревью после исправления ошибок.

olga-rks committed:
Изменила переменную --rgb на --opacity.

Повторная отправка проектной работы после исправления ошибок.

Avaakum reviewed (approved):
Развернуть общий комментарий ⬇︎

Здравствуйте! Рад вам сообщить, что ваша работа принята. Сейчас страничка выглядит прекрасно, всё работает отлично. Отличная семантика и структура. Желаю вам удачи в следующих спринтах и проектах! 




Ссылки:

- Можно использовать неразрывные пробелы для того чтобы текст не переносился в неожиданных местах. Неразрывные пробелы заменяют обычные, если нужно чтобы какая-то фраза не переносилась, например не нужно чтобы "на собаках" раскидывалось по разным строчкам при некоторых ширинах экранов, тогда `«на&nbsp;собаках»`. Но если страница с адаптацией, то всегда стоит проверять как текст с пробелами выглядит на всех ширинах.
- Про альтернативный текст для картинок https://habr.com/ru/post/341810/
- [Ссылка про подгрузку шрифтов](https://only-to-top.ru/blog/tools/2019-03-13-nastrojte-pokaz-vsego-teksta-vo-vremya-zagruzki-veb-shriftov.html) 



Несколько  ссылок на материал по теме нативных CSS переменных от авторов практикума:

- [1 часть](https://tproger.ru/articles/custom-properties-in-css-part-1/)

- [2 часть](https://tproger.ru/articles/custom-properties-in-css-part-2/)

- [3 часть](https://tproger.ru/articles/custom-properties-in-css-part-3/)
